File: packages/apollos-ui-media-player/ios/ApollosPlayer.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(ApollosPlayer, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a withB:(float)b
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(isPictureInPictureSupported:(RCTPromiseResolveBlock)resolve
                  withRejecter:(RCTPromiseRejectBlock)reject)


@end

File: packages/apollos-ui-media-player/ios/ApollosPlayer.swift
import AVKit

@objc(ApollosPlayer)
class ApollosPlayer: NSObject {

    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(a*b)
    }
    
    @objc(isPictureInPictureSupported:withRejecter:)
    func isPictureInPictureSupported(
        resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
        resolve(AVPictureInPictureController.isPictureInPictureSupported());
    }
}
